### DISCUSSION TITLE
VBUS data in tasmota-SML not complete?

### DISCUSSION THREAD

**Discussion by juergiboy** [arendst/Tasmota#17895] [Q&A] [answered]
Hi there,
i have successfully connected my Resol solar (and heating) controller Deltasol-M to Tasmota-SML to read the data via VBUS.

**First of all my respect for this great implementation in Tasmota!**

I can read the data from the sensors without any problems.
But it seems to me that not all data may be available, according to the protocol header 17 (0x11) data frames are to be transmitted - but apparently only 7 or 8 are made available:

```
20:42:35.918 : aa 10 00 11 73 10 00 01 11 49 1b 00 49 00 04 17 0d 02 47 00 04 25 4d 00 38 22 05 53 38 22 38 22 05 46 18 01 24 00 01 41
20:42:36.139 : aa 10 00 12 73 10 00 01 03 56 18 01 24 00 01 41 21 00 3d 01 04 1c 3e 01 0b 00 01 34
20:42:36.359 : aa 10 00 16 73 10 00 01 03 52 4d 00 47 00 05 66 00 00 0e 00 04 6d 03 01 00 00 01 7a
20:42:37.606 : aa 00 00 11 73 20 00 05 00 00 00 00 00 00 00 56
20:42:37.919 : aa 10 00 11 73 10 00 01 11 49 1b 00 49 00 04 17 0d 02 47 00 04 25 4d 00 38 22 05 53 38 22 38 22 05 46 0d 01 24 00 01 4c
20:42:38.138 : aa 10 00 12 73 10 00 01 03 56 0d 01 24 00 01 4c 21 00 3d 01 04 1c 3e 01 0b 00 01 34
20:42:38.359 : aa 10 00 16 73 10 00 01 03 52 4d 00 47 00 05 66 00 00 0e 00 04 6d 03 01 00 00 01 7a
20:42:39.606 : aa 00 00 11 73 20 00 05 00 00 00 00 00 00 00 56
```
What can i do to get the rest of my data?
Thanks.

**Comment by barbudor** [comment]
Please share more details about the device, including documentation that says 17 frames should be sent
As well as your Tasmota configuration/script etc 
If Tasmota display only 8 it may be because you need to do something to get them all

Or because the data is arriving too fast and some got lost so details like bitrate and amount of data sent would be usefull

**Reply by juergiboy** [reply]
Here is the specification of the Resol-VBUS: [https://hobbyelektronik.org/w/images/0/04/VBus-Protokollspezifikation.pdf](url)
Byte 9 of the header contains the number of data frames.

**Reply by barbudor** [reply]
English version ?

**Reply by juergiboy** [reply]
Oh, I'am sorry, here is it in english: [https://github.com/epenet/pyvbus/blob/master/documentation/VBus%20Protocol%20Specification%20-%20English%202011-01-27.pdf](url)
You can see the VBUS header at page 14 and the specification for my controller at page 28.

**Comment by gemu2015** [comment, marked as answer]
i assume you serial buffer is too small, assign a larger buffer, either 512 or even more bytes
1,=so3,512

**Reply by juergiboy** [reply]
Thanks for the hint, but this doesn't change anything unfortunately.
The system (sensor53 d1) returns only 30 bytes after the CRC-byte from the VBUS-header.

**Reply by gemu2015** [reply]
What version do you use?
Older versions need a #define for larger serial buffers

**Reply by juergiboy** [reply]
I use the version 12.3.1.6

**Reply by gemu2015** [reply]
ok, then enlarge all serial related buffers, including dump buffer
1,=so3,512,512,512

**Reply by juergiboy** [reply]
Ok, thats it, it works!
Thanks!!
